analytics/models.py

```diff
@@ -0,0 +1,67 @@
+import enum
+from sqlalchemy import Integer, Column, Enum, String, DateTime
+from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.sql import func
+from sqlalchemy.ext.declarative import declarative_base
+
+Base = declarative_base()
+
+
+class SearchEvent(Base):
+    """
+    Store searches linked to a session UUID.
+    """
+    __tablename__ = "search_event"
+
+    id = Column(Integer, primary_key=True)
+    date = Column(DateTime, server_default=func.now(), index=True)
```

Comment:
Could we call this `timestamp` or similar to indicate that it is a datetime and not a date? Same applies to fields below.

Also, it might make sense to put `id` and `timestamp` in a base class to avoid repeating that code.

Reply:
I renamed `date` -> `timestamp` and moved the `id` and `timestamp` fields into a separate mixin class, the result is less cluttered